### PR TITLE
 Add object's own position for each collision to moveresult 

### LINKED ISSUE #9966: Moveresult giving collision position
## Problem
Moveresult does not give collision position of the entity, so, for example: My specific use is moving an arrow back to the position that it originally hit the node to stop it from sliding around.

## Solutions

Just dump collision_position = v3f(x,y,z) into the moveresult return

### PR DESCRIPTION
**Pull request by grorp**
For reasoning and use cases, see https://github.com/minetest/minetest/issues/9966, https://github.com/minetest/minetest/pull/9978#issuecomment-637705331, https://github.com/minetest/minetest/pull/9978#issuecomment-654981567. Closes #9966 

## To do

This PR is a  Ready for Review.

## How to test

Try this modpack: https://github.com/grorp/grenade-something

Throw grenades at a player and see that they explode exactly where they hit.

### CODE REVIEW

File: src/script/common/c_content.cpp [hunk 1]
```diff
@@ -2528,6 +2528,9 @@ void push_collision_move_result(lua_State *L, const collisionMoveResult &res)
 			lua_setfield(L, -2, "object");
 		}
 
+		push_v3f(L, c.new_pos / BS);
+		lua_setfield(L, -2, "new_pos");
```
Comment (sfan5) [inline src/script/common/c_content.cpp:2532]: need to change `lua_createtable` to 0, 6 above

Comment (grorp) [reply src/script/common/c_content.cpp:2532]: true

**Review by sfan5 — APPROVED** [review]
works perfectly

**Comment by wsor4035** [comment]
seems `new_pos` in `moveresult` should be added to lua_api.md

**Comment by appgurueu** [comment]
> seems `new_pos` in `moveresult` should be added to lua_api.md

[Isn't it already?](https://github.com/minetest/minetest/pull/14608/files#diff-c60ce2a4d38acfedec97ac1b19e9e207aa60f223f1b7fdd1048c00bdf1f8d5d9)

**Comment by wsor4035** [comment]
> > seems `new_pos` in `moveresult` should be added to lua_api.md
> 
> [Isn't it already?](https://github.com/minetest/minetest/pull/14608/files#diff-c60ce2a4d38acfedec97ac1b19e9e207aa60f223f1b7fdd1048c00bdf1f8d5d9)

appears im just blind, saw the feature entry, but missed the above.

**Pull request merged**